Fix up requirements.txt

blinkdog opened the pull request:
Fixes `requirements.txt` to include the things this action actually needs.

@ric-evans Can you review the `action.yml` and `generate_matrix.py` also? Just want to make sure I'm not overlooking something obvious, especially in the `action.yml`.

ric-evans commented:
You aren't using your `requirements.txt`: https://github.com/WIPACrepo/wipac-dev-mongo-versions-action/blob/fix-requirements/action.yml#L35

ric-evans commented:
You may want to consider using exponential backoff by using a for-loop (i.e. `time.sleep((i+1)*10)`) with an upper limit.

ric-evans commented:
You'd save some parsing if you kept the `Version` objects as the keys in `grouped`: https://github.com/WIPACrepo/wipac-dev-mongo-versions-action/blob/fix-requirements/generate_matrix.py#L35-L44

ric-evans commented:
The pre-release logic is cool, but you may want to warn the user `echo ::warn:: ...` that they are testing a non-stable version (things could fail?). https://github.com/WIPACrepo/wipac-dev-mongo-versions-action/blob/fix-requirements/generate_matrix.py#L46-L50

This may also be better suited as an optional configuration, i.e. a new input: `allow_prereleases`.

ric-evans commented:
Add logging for https://github.com/WIPACrepo/wipac-dev-mongo-versions-action/blob/fix-requirements/generate_matrix.py#L24-L29?

blinkdog commented:
> I left my reviews above -- this'll be a useful action

Excellent. Thank you, I'll incorporate the feedback as soon as I can.

blinkdog commented:
> You aren't using your `requirements.txt`: https://github.com/WIPACrepo/wipac-dev-mongo-versions-action/blob/fix-requirements/action.yml#L35

Good catch; fixed.

> You may want to consider using exponential backoff by using a for-loop (i.e. `time.sleep((i+1)*10)`) with an upper limit.

This sleep is just a self-imposed rate limit (1 request/sec) against Docker Hub; just to be polite to their API endpoint.
If a request fails, `r.raise_for_status()` and fail out of there.
Because we completely back off, there's no need for exponential back off.

> The pre-release logic is cool, but you may want to warn the user `echo ::warn:: ...` that they are testing a non-stable version (things could fail?). https://github.com/WIPACrepo/wipac-dev-mongo-versions-action/blob/fix-requirements/generate_matrix.py#L46-L50
> 
> This may also be better suited as an optional configuration, i.e. a new input: `allow_prereleases`.

I added a new input `omit_prereleases` to allow pre-releases to be omitted from the output.
